### PR TITLE
Unzip files

### PR DESCRIPTION
**Pull request by tir-bogo**
+ Made refactoring of convert_files
+ Methods has defined types end return types
+ UnzipFiles class

### CODE REVIEW

File: logfile/operations/types/convert_files.py [hunk 1]
```diff
@@ -21,129 +24,71 @@ class ConvertFiles(OperationBase):
...
-        """
-        return self.config[self.__new_file_extension_instruction_key]
-
-    @staticmethod
-    def __list_item_contains_string(arr, item):
-        """
+            bool: Item is found
         """
         for val in arr:
             if val in item:
```
Comment (pkhbogo) [inline logfile/operations/types/convert_files.py:61]: Shouldn't it be item in val?

File: logfile/operations/types/unzip_files.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            tar.extractall(extract_to)
+            tar.close()
+
+            logging.debug("Deleting file '{filepath}'")
+            Path(filepath).unlink()
+
+            return extract_to
+
+        except OSError as exc:
+            logging.warning(f"Error while extracting file '{filepath}' {exc}")
```
Comment (pkhbogo) [inline logfile/operations/types/unzip_files.py:56]: What happens when you get an exception?

**Review by pkhbogo — APPROVED** [review]
Few minor things.

File: logfile/operations/types/convert_files.py (hunk 1 above)
Comment (tir-bogo) [reply logfile/operations/types/convert_files.py:61]: This is correct but the documentation was wrong. I fixed it

File: logfile/operations/types/unzip_files.py (hunk 2 above)
Comment (tir-bogo) [reply logfile/operations/types/unzip_files.py:56]: I print a warning.

Comment (pkhbogo) [reply logfile/operations/types/unzip_files.py:56]: Yeees....., but what happens after that? ;-)

File: logfile/operations/types/convert_files.py (hunk 1 above)
Comment (pkhbogo) [reply logfile/operations/types/convert_files.py:61]: OK, but then I think the name is misleading. What about `__list_item_in_string`

**Review by pkhbogo — COMMENTED** [review]
I still think, you need to change a few things.

File: logfile/operations/types/unzip_files.py (hunk 2 above)
Comment (tir-bogo) [reply logfile/operations/types/unzip_files.py:56]: If I cant extract it  (the file is corrupted) i cant do anything about it, so i just continue.

**Pull request merged**